pelican/log.py

```diff
@@ -87,15 +87,24 @@ class FatalLogger(LimitLogger):
     warnings_fatal = False
     errors_fatal = False
 
-    def warning(self, *args, **kwargs):
-        super().warning(*args, **kwargs)
+    def callHandlers(self, record):
+        """
+        Raise RuntimeError if log level is higher than acceptable.
+
+        Level limit set via the ``--fatal`` flag when running pelican.
+        """
+        super().callHandlers(record)
+        level_limit = None
+
         if FatalLogger.warnings_fatal:
-            raise RuntimeError('Warning encountered')
+            level_limit = logging.WARNING
+        elif FatalLogger.errors_fatal:
+            level_limit = logging.ERROR
+        else:
+            return
 
-    def error(self, *args, **kwargs):
-        super().error(*args, **kwargs)
-        if FatalLogger.errors_fatal:
-            raise RuntimeError('Error encountered')
+        if record.levelno == level_limit:
```

Comment:
Shouldn't this be:
```python
if record.levelno >= level_limit:
```

Reply:
Good catch, Thanks!  I feel like that's what I had before shuffling a bunch of code around :(
Time to write up some tests, it seems...

Reply:
Tests would be excellent. I was going to suggest adding some tests since this part wasn't tested originally but I didn't want to pile it on this PR and create extra work :).